dio/test/interceptor_test.dart

```diff
@@ -349,6 +349,40 @@ void main() {
         expect(response.requestOptions.contentType, isNull);
       });
 
+      test("Interceptor gets stacktrace in onError", () async {
+        final dio = Dio();
+        dio.options.baseUrl = EchoAdapter.mockBase;
+        dio.httpClientAdapter = EchoAdapter();
+
+        late final StackTrace caughtStackTrace;
+        dio.interceptors.addAll([
+          InterceptorsWrapper(
+            onError: (err, handler) {
+              caughtStackTrace = err.stackTrace!;
+              handler.next(err);
+            },
+          ),
+          InterceptorsWrapper(onRequest: (options, handler) {
+            final error = DioError(
+              error: Error(),
+              requestOptions: options,
+            );
+            handler.reject(error, true);
+          }),
+        ]);
+
+        await expectLater(
+          dio.get('/error'),
+          throwsA(predicate(
+            (error) {
+              return error is DioError &&
+                  caughtStackTrace.toString() == error.stackTrace.toString();
```

Comment:
```suggestion
                  caughtStackTrace == error.stackTrace;
```

Reply:
The problem I got when trying to do proper equal check is that it seems like the StackTrace objects are two different instances, and equality check is just checking the instances.

So checking the output of `toString` was the easiest way for me to check the contents of the two StackTraces against each other.
Is there a better way for us to solve this?

Reply:
I'm not aware of any, but then using toString is fine for me